### PR TITLE
DaemonSet template

### PR DESCRIPTION
**Pull request by adamancini**

### CODE REVIEW

**Comment by adamancini** [comment]
Deployed with a simple nginx daemonset

```
nginx:
  enabled: true
  type: daemonset
  containers:
    nginx:
      image:
        repository: docker.io/nginx
        tag: latest
```

**Comment by adamancini** [comment]
and I added the `app` class and moved deploy, daemonset, and statefulset into class templates

**Comment by chris-sanders** [comment]
Oh you're right, I see that now looking at the configmap pattern. That's a bit confusing to me because you have templates like _names and _env_vars _annocations which aren't top level keys to be iterated over. But I do see there are several others in `lib` that as you point out are ranging over a key and call `class` templates.

I guess that's fine, let me think about that for a bit.

File: charts/replicated-library/templates/lib/_apps.tpl [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{{/*
+Renders the app objects into Deployments, DaemonSets, and StatefulSets as required by the chart.
+*/}}
+{{- define "replicated-library.apps" -}}
+  {{- range $name, $app := .Values.apps }}
```
Comment (chris-sanders) [inline charts/replicated-library/templates/lib/_apps.tpl:5]:
```suggestion
  {{- range $name, $appValues := .Values.apps }}
```
Maybe we can just drop this extra mapping and go straight to $appValues?

File: charts/replicated-library/templates/lib/_apps.tpl [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    {{- if $app.enabled -}}
+      {{- $appValues := $app -}}
```
Comment (chris-sanders) [inline charts/replicated-library/templates/lib/_apps.tpl:7]:
```suggestion

```

**Review by chris-sanders — CHANGES_REQUESTED** [review]
One minor nit, and I know you've copied this from the previous _all but I think we can reduce the variables by one making it a bit easier to read/follow. Other than that, this looks good to me.

File: charts/replicated-library/templates/lib/_apps.tpl (hunk 1 above)
Comment (adamancini) [reply charts/replicated-library/templates/lib/_apps.tpl:5]: yeah that works

**Pull request merged**